### DISCUSSION TITLE
Contact Timeline

### DISCUSSION THREAD

**Discussion by ux-fran** [nl-design-system/backlog#351] [Component suggestions]
<img width="480" alt="visuele weergave van een contact timeline component" src="https://github.com/user-attachments/assets/28d719f6-bdc6-482b-b2a3-546042386592" />

We willen graag meer leren over de Contact Timeline component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1729189457717759})

---

## Naam

Contact Timeline

## Doel

Toont een chronologische lijst van contactmomenten, met de mogelijkheid om aanvullende details per contactmoment te bekijken.

## Onderzoek

[Generieke Services schetsboek - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=11-6995&p=f&t=6N0ZfdQewPcatf8v-11)
In dit Figma schetsboek is inspiratie en schetsen te vinden die het ontwerpproces van de Contact Timeline component weergeven. Het bevat verschillende use cases, zoals 'contact button', ‘reactie medewerker’ en ‘verstreken deadline’.

[Generieke Services onderzoek - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=1707-140361&p=f&t=o94bkjhamnmKxoE3-11)
In Figma is informatie verzameld vanuit verschillende organisaties, die inzichten bieden in manieren van notificeren. Dit omvat onder andere hoe een ‘contactmoment’ (oftewel de Contact Timeline component) technisch geïmplementeerd.

[Generieke Services concept - Gemeente Den Haag](https://www.figma.com/design/6CFtgDTS2oUceGqTWYXhdN/%F0%9F%95%B9%EF%B8%8F-Generieke-Services---MijnDenHaag?node-id=714-33964&t=6N0ZfdQewPcatf8v-11)
In Figma is uitgebreide informatie beschikbaar over het contactmoment: hoe deze eruitziet, hoe deze werkt, en verschillende scenario’s waarin de Contact Timeline component wordt gebruikt. Denk bijvoorbeeld aan de inhoud van de Contact Timeline, en waar deze zich bevindt in de 'Mijn omgeving'.
 
[Rapport Generieke Services 2023 - Gemeente Den Haag (PDF, 9 MB)](https://github.com/user-attachments/files/16993296/Rapport.generieke.services.23032023.def.pdf)
Uit dit gebruikersonderzoek blijkt dat de Contact Timeline goed functioneert. Enkele inzichten:

- Het overzicht van contactmomenten wordt als erg waardevol ervaren.
- Er is behoefte om eerdere contactmomenten eenvoudig terug te kunnen vinden.
- De iconen bij contactmomenten die het kanaal van contact aanduiden, zijn vervangen door woorden, aangezien niet alle iconen begrepen werden.
- Door het onderwerp van het contactmoment te verrijken met informatie die ook in de notificatiemail staat, is het nu nog duidelijker waar het contactmoment over gaat.
- Op mobiel is het niet meteen duidelijk wat het overzicht van contactmomenten inhoudt; op desktop lijkt dit makkelijker te interpreteren.
- Men verwijst naar het overzicht als een ‘omgekeerde tijdlijn’, wat in tegenstelling staat tot het statusoverzicht dat bovenaan de pagina wordt weergegeven.

[Onderzoek Generieke Services - Miro](https://miro.com/app/board/uXjVP0OE2fg=/?share_link_id=511269957959)
Miro board dat is gebruikt tijdens het gebruikersonderzoek waarin inzichten te vinden zijn over de Contact Timeline.

[Timeline - Ant Design](https://ant.design/components/timeline)
Timeline component bij Ant Design.

[Timeline - Material UI](https://mui.com/material-ui/react-timeline/)
Timeline component bij Material UI gemaakt in React.

[Activity Timeline - Salesforce](https://www.lightningdesignsystem.com/components/activity-timeline/)
Activity Timeline component bij Saleforce Design System.

**Comment by ux-fran** [comment]
## Research nov 2024
Gekeken in diverse files van Amsterdam, DSO, Open formulieren, Open Gemeenten, Gemeente Rotterdam, RVO, Gemeente Utrecht en daar verder iets vergelijkbaars gevonden.

Heb ik wat gemist of weet jij (de lezer) misschien nog een organisatie of andere gemeente met een vergelijkbaar component, vul gerust aan!

**Comment by ux-fran** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/timeline)

---

### Screenshots / Varianten

Standaard tijdlijn
Gebruik de standaard tijdlijn als je per gebeurtenis slechts een datum en een actie wilt tonen. Deze tijdlijn biedt de gebruiker snel inzicht in het gehele proces, met eventueel doorklikmogelijkheid tot meer informatie. Deze variant kan inactieve items bevatten.
<img width="680" alt="image" src="https://github.com/user-attachments/assets/923f0866-01dd-4466-a9aa-bef9dd53af56" />

Compacte tijdlijn
Gebruik de compacte tijdlijn als je de huidige status van de gebruiker in een proces wilt tonen. De gepasseerde en toekomstige gebeurtenissen worden ook meegenomen in de tijdlijn, maar worden anders opgemaakt. Deze tijdlijn is niet interactief.
<img width="680" alt="image" src="https://github.com/user-attachments/assets/d3a1b099-0474-4ab9-9dd1-16a88fbe52df" />

Grote tijdlijn
Wil je meer informatie kwijt dan in de standaard tijdlijn? Overweeg dan een grote tijdlijn. Hierin is ruimte voor een datum, een titel, een uitleg en een “Lees meer” link. Deze tijdlijn kan echter geen inactieve items bevatten.
<img width="680" alt="image" src="https://github.com/user-attachments/assets/475b32e2-1274-4556-8fd0-fcc5b6ea36ec" />

Responsive gedrag
<img width="280" alt="image" src="https://github.com/user-attachments/assets/01317fef-ffae-4c31-931d-b516b7469220" /><br>

<img width="280" alt="image" src="https://github.com/user-attachments/assets/bb4f0211-56b4-4a9b-9098-af159b80ef1b" /><br>

<img width="280" alt="image" src="https://github.com/user-attachments/assets/8b79d3e4-57c5-4ce5-9782-08b4a08dc783" />

---


### Notities

- Onder de naam: Timeline.
- Nieuwste datum staat bovenaan. 
- Als de huidige status wordt aangegeven, dan staat het meest recente item bovenaan en het oudste onderaan.
- De datum bevat zowel een dag en maand, genoteerd als: 12 december, maar kan ook een reeks weergeven, zoals ‘1 mei t/m 30 augustus’.
- Niet duidelijk of het jaar wordt vermeld wanneer de datum zich over twee jaren uitstrekt.
- De datum is in code een header. Welke dat moet zijn is afhankelijk van waar in de hiërarchie je zit. 
- Standaard tijdlijn: onder iedere datum staat een Link, tenzij deze 'uitgeschakeld' (disabled) is.
- Kan uit Badges bestaan.
- Compacte tijdlijn: bevat enkel Headings. Vertoont een status.
- Grote tijdlijn: bevat data, een Link met daaronder een mogelijke description.
- Headings of Links zijn redelijk kort beschreven. 
- Richtlijnen beschikbaar.

**Comment by ux-fran** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-contact-timeline--docs)
- [Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=10817-61351&t=uz0TWRngCXR1NFzv-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/ContactTimeline)

---

### Screenshots / Varianten

Default
<img width="700" alt="image" src="https://github.com/user-attachments/assets/1c3d74c5-8fd9-4f05-9f4d-4602718bffd2" />

Collapsible
<img width="700" alt="image" src="https://github.com/user-attachments/assets/4dddd923-6b54-4a65-984a-594532c31383" />

Collapsible with expanded item
<img width="700" alt="image" src="https://github.com/user-attachments/assets/64d898b6-1315-4241-a5f1-a9e882c3e4d6" />

Without sender
<img width="700" alt="image" src="https://github.com/user-attachments/assets/915965bc-bdf9-45f7-9333-0ad7ae360c04" />

With file
<img width="700" alt="image" src="https://github.com/user-attachments/assets/30c160e6-6f8b-462e-9728-42ed782ac4f6" />

Responsive gedrag
<img width="320" alt="image" src="https://github.com/user-attachments/assets/7aa27e9e-f6f2-4666-a11f-2c6260ae6371" /><br>

<img width="320" alt="image" src="https://github.com/user-attachments/assets/d6ad027a-d2d4-41da-8adc-7baec335f56e" /><br>

<img width="320" alt="image" src="https://github.com/user-attachments/assets/d979e436-b371-4776-8102-1cbf64903d5f" />

---

### Notities
- Onder de naam: Contact Timeline.
- De contactmomenten zijn gesorteerd op datum, met het meest recente contactmoment bovenaan en het oudste onderaan.
- Elk contactmoment bevat een titel, datum en kanaal, en kan bestaan uit een afzender, bestand en (tekstuele) inhoud.
- De datum bestaat uit de dag, maand en jaar, genoteerd als: 12-12-2024.
- Wanneer de datum gisteren of vandaag is, wordt in plaats van een exacte datum de term ‘gisteren’ of ‘vandaag’ gebruikt.
- Kan worden uit- en ingeklapt. Bij het uitklappen zie je de afzender, het bestand en de tekstuele inhoud.
- Titels zijn redelijk kort beschreven.
- Richtlijnen beschikbaar (Figma).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Contact Timeline op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=contact+timeline)
